src/framework/MOM_random.F90

```diff
@@ -46,6 +47,28 @@ real function random_01(CS)
 
 end function random_01
 
+!> Returns a random number between 0 and 1
+!! See https://arxiv.org/abs/2004.06278. Not an exact reproduction of "squares" because Fortran
+!! doesn't have a uint64 type, and not all compilers provide integers with > 64 bits...
+real function random_01_CB(ctr, key)
+  integer, parameter   :: int64 = selected_int_kind(10) ! Integer with >= 64 bits
```

Comment:
Thank you for this contribution. Please doxygenize the arguments of this subroutine following the [MOM6 convention](https://github.com/NOAA-GFDL/MOM6/wiki/Doxygen). You can use subroutine random_2d_01 as an example.

Reply:
Thanks Marshall and Gustavo. I've switched to in64 from iso_fortran_env, and added the docstrings.